python/src/worker/tasks/background_expressed_genes.py

```diff
@@ -0,0 +1,128 @@
+import json
+
+import backoff
+import requests
+from aws_xray_sdk.core import xray_recorder
+
+from ..config import config
+from ..helpers.find_cell_ids_in_same_hierarchy import (
+    find_all_cell_ids_in_cell_sets, find_cell_ids_in_same_hierarchy)
+from ..helpers.find_cells_by_set_id import find_cells_by_set_id
+from ..helpers.remove_regex import remove_regex
+from ..helpers.s3 import get_cell_sets
+from ..result import Result
+from . import Task
+
+
+class GetBackgroundExpressedGenes(Task):
+    def __init__(self, msg):
+        super().__init__(msg)
+        self.experiment_id = config.EXPERIMENT_ID
+        self.pagination = {}
+
+    def _format_result(self, result):
+        # Return a list of formatted results.
+        return Result({"genes": result["genes"]})
+
+    def _format_request(self):
+        # get cell sets from database
+        resp = get_cell_sets(self.experiment_id)
+
+        first_cell_set_name = self.task_def["cellSet"]
+        second_cell_set_name = self.task_def["compareWith"]
+        basis = self.task_def["basis"]
+
+        # Check if the comparsion is between all the cells or within a cluster
+        if not basis or ("all" in basis.lower()):
+            # In case that we are not going to filter by a cluster we remain the set
+            #  empty
+            filtered_set = set()
+        else:
+            # In the case that we filter by a cluster we keep the cells id in a set
+            filtered_set = set(find_cells_by_set_id(basis, resp))
+
+        # mark cells of first set
+        first_cell_set = find_cells_by_set_id(first_cell_set_name, resp)
+
+        # mark cells of second set
+        # check if the second set is composed by the "All other cells"
+        if (
+            second_cell_set_name == "background"
+            or "all" in second_cell_set_name.lower()
+        ):
+            # Retrieve all the cells (not necessary at the same hierachy level)
+            complete_cell_set = set(find_all_cell_ids_in_cell_sets(resp))
+            # Filter with those that are not in the first cell set
+            second_cell_set = [
+                item for item in complete_cell_set if item not in first_cell_set
+            ]
```

Comment:
this could be made faster using sets... complete cell_set is already a set. I think it does not preserve order though. Do we need order for any of this?

Other possible concern might be uniqueness of cell_ids. sets only contain unique elements. I think this might not be a problem though, since we are converting everything to sets everywhere.

Reply:
according to [this SO answer](https://stackoverflow.com/a/41125957), it goes from O(n) to roughly O(1). It might be worth it, since cell sets can be arbitrarily large.

```suggestion
            second_cell_set = list(complete_cell_set.difference(first_cell_set))
```

We could remove the `list` call if we just need a set. but left it since it was a list comprehension.

Reply:
changed to use set operation